packages/mantine/src/components/table/__tests__/TableActions.spec.tsx

```diff
@@ -151,6 +143,56 @@ describe('Table.Actions', () => {
         expect(screen.queryByRole('button', {name: 'Eat vegetable'})).not.toBeInTheDocument();
     });
 
+    it('keeps the selected row data in sync with the data prop', async () => {
+        // When a row is selected and the data prop changes, the selected row data should be updated with the new data without having to reselect the row
+        type Food = {name: string; status: 'fresh' | 'eaten'};
+        const foodColumnHelper = createColumnHelper<Food>();
+        const myColumns: Array<ColumnDef<Food>> = [foodColumnHelper.accessor('name', {enableSorting: false})];
+        const user = userEvent.setup();
+
+        const Fixture = () => {
+            const [data, setData] = useState<Food[]>([{name: 'fruit', status: 'fresh'}]);
+            const eatFruit = () => setData([{name: 'fruit', status: 'eaten'}]);
+
+            const store = useTable<Food>();
+            return (
+                <Table<Food>
+                    store={store}
+                    data={data}
+                    columns={myColumns}
+                    getRowActions={(selected: Food[]) => [
+                        {
+                            group: '$$primary',
+                            component:
+                                selected[0].status === 'fresh' ? (
+                                    <Table.ActionItem key="eat" onClick={eatFruit}>
+                                        Eat
+                                    </Table.ActionItem>
+                                ) : null,
+                        },
+                        {
+                            group: '$$primary',
+                            component:
+                                selected[0].status === 'eaten' ? (
+                                    <Table.ActionItem key="trash">Throw away</Table.ActionItem>
+                                ) : null,
+                        },
+                    ]}
```

Comment:
Any reason to have two different actions?

```suggestion
                    getRowActions={(selected: Food[]) => [
                        {
                            group: '$$primary',
                            component:
                                selected[0].status === 'fresh' 
                                ? (
                                    <Table.ActionItem key="eat" onClick={eatFruit}>
                                        Eat
                                    </Table.ActionItem>
                                ) 
                                : <Table.ActionItem key="trash">Throw away</Table.ActionItem>,
                        },
                    ]}
```

Reply:
It was to better mimic the reality of the admin 👾 You will often have both actions but protected by guards rather than conditionally pushing into the array. I can do what you suggest if you prefer.

Reply:
I don't mind either solutions, go with what you prefer. In the admin it could be a Guard with a fallback which would be closer to what I propose